### PR TITLE
module_utils.postgres: added unittests

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY
1. Unit tests for module_utils.postgres
2. Using LooseVersion instead of string comparison

+label: postgresql
+label: databases

##### ISSUE TYPE
- Bugfix Pull Request

### CODE REVIEW

**Comment by Andersson007** [comment]
ready_for_review

File: lib/ansible/module_utils/postgres.py [hunk 1]
```diff
@@ -110,22 +113,24 @@ def connect_to_db(module, autocommit=False, fail_on_conn=True, warn_db_default=T
...
         # Switch role, if specified:
-        cursor = db_connection.cursor(cursor_factory=DictCursor)
         if module.params.get('session_role'):
+            cursor = db_connection.cursor(cursor_factory=psycopg2.extras.DictCursor)
             try:
                 cursor.execute('SET ROLE %s' % module.params['session_role'])
             except Exception as e:
                 module.fail_json(msg="Could not switch role: %s" % to_native(e))
-        cursor.close()
+            finally:
```
Comment (andytom) [inline lib/ansible/module_utils/postgres.py:142]: 👍, this is a nice little refactor.

**Comment by andytom** [comment]
LGTM

**Comment by Andersson007** [comment]
@andytom , thank you for reviewing!
I changed yield_fixture pytest decorators to just fixture by last commit because it is depricated (thanks @mattclay for pointing out).

**Comment by Andersson007** [comment]
@mattclay , it is shipit'ed by andytom, a member of our Postgres team, could you merge it please?

After that I can remove ensure_libs (depricated function) and its test file test/units/module_utils/test_postgresql.py from the source code by particular PR.

**Comment by andytom** [comment]
shipit

**Comment by Andersson007** [comment]
ready_for_review

**Comment by andytom** [comment]
shipit

**Comment by Andersson007** [comment]
@andytom , thank you for reviewing 
@mattclay , thank you for reviewing and merging.
Feel free if you need some reviews or other help

**Pull request merged**